packages/expo-modules-core/android/src/main/java/expo/modules/adapters/react/views/ViewManagerAdapterUtils.java

```diff
@@ -54,4 +54,11 @@ public class ViewManagerAdapterUtils {
       }
     }
   }
+
+  public static String normalizeEventName(final String eventName) {
+    if (eventName.startsWith("on")) {
+      return "top" + eventName.substring(2);
```

Comment:
I'm going to do this a little bit differently on iOS. The "top" prefix is something that RN adds for no reason and they may change it in their code, so I would prefer not to follow their path 😅 Take a look at other event emitters, e.g. here https://github.com/facebook/react-native/blob/d35aab45b9a2cb79f234fd0d599cc50793866ee2/ReactCommon/react/renderer/components/scrollview/ScrollViewEventEmitter.cpp#L68-L71
where the event name is without "on" and "top". I think it's better to do the same here and let https://github.com/facebook/react-native/blob/d35aab45b9a2cb79f234fd0d599cc50793866ee2/ReactCommon/react/renderer/core/EventEmitter.cpp#L25-L32 handle this name later. What do you think?

```suggestion
      return eventName.substring(2).decapitalize();
```

Reply:
it's little different. for dispatching event, we could use the event name without *top* prefix, as the [ScrollViewEventEmitter](https://github.com/facebook/react-native/blob/d35aab45b9a2cb79f234fd0d599cc50793866ee2/ReactCommon/react/renderer/components/scrollview/ScrollViewEventEmitter.cpp#L68-L71) you shared. however, to declare supported event types for a component, the *top* prefix is needed, e.g. the [scrollview event declarations](https://github.com/facebook/react-native/blob/d35aab45b9a2cb79f234fd0d599cc50793866ee2/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ScrollEventType.java#L21), [another example from codegen for ios](https://github.com/facebook/react-native/blob/d35aab45b9a2cb79f234fd0d599cc50793866ee2/Libraries/Components/ScrollView/ScrollViewNativeComponent.js#L93). 

i don't like the *top* prefix either. if we are going to remove the top prefix, i think we should customize an event emitter from c++ side in fabric, that would introduce more complexity.